sklearn/linear_model/_ridge.py

```diff
@@ -1616,6 +1616,17 @@ def fit(self, X, y, sample_weight=None):
             self.alpha_ = gs.best_estimator_.alpha
             self.best_score_ = gs.best_score_
 
+        min_alpha = min(self.alphas)
+        max_alpha = max(self.alphas)
+        if self.alpha_ in [min_alpha, max_alpha]:
+            warnings.warn('The optimal value for the regularization '
+                          'parameter ''alpha'' was %g which '
+                          'lies at a boundary of the explored range '
+                          '(between %g and %g). Consider setting '
+                          'the ''alphas'' parameter to explore a '
+                          'wider range.'
+                          % (self.alpha_, min_alpha, max_alpha))
```

Comment:
Perhaps this should be a ConvergenceWarning, since like in cases of non-convergence, we do not believe that the current parameters allowed us to find an optimal solution.